src/utils_general/raster_manipulation.py

```diff
@@ -99,111 +100,88 @@ def fix_calendar(ds, timevar="F"):
+    region will be included when computing the stat.
+    If True all cells touching the region will be included.
+    :param geom_col: name of the column in boundary_path
+    containing the polygon geometry
+    :return: dataframe containing the computed statistics
     """
-    df = gpd.read_file(boundary_path)
-    # TODO: decide if we want to upsample and if yes, implement if
-    # upscale_factor: forecast_array, transform =
-    # resample_raster(raster_path, upscale_factor) else:
-
-    # extract statistics for each polygon. all_touched=True includes all
-    # cells that touch a polygon, with all_touched=False only those with
-    # the center inside the polygon are counted.
-    df["max_cell"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            band=band,
-            nodata=nodata,
-        )
-    )["max"]
-    df["max_cell_touched"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            all_touched=True,
-            band=band,
-            nodata=nodata,
-        )
-    )["max"]
-
-    df["avg_cell"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            band=band,
-            nodata=nodata,
-        )
-    )["mean"]
-    df["avg_cell_touched"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            all_touched=True,
-            band=band,
-            nodata=nodata,
-        )
-    )["mean"]
-
-    # calculate the percentage of the area within an geographical area
-    # that has a value larger than threshold
-    forecast_binary = np.where(raster_array >= threshold, 1, 0)
-    bin_zonal = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=forecast_binary,
-            affine=raster_transform,
-            stats=["count", "sum"],
-            band=band,
-            nodata=nodata,
-        )
-    )
-    df["perc_threshold"] = bin_zonal["sum"] / bin_zonal["count"] * 100
-    bin_zonal_touched = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=forecast_binary,
-            affine=raster_transform,
-            all_touched=True,
-            stats=["count", "sum"],
-            band=band,
-            nodata=nodata,
-        )
-    )
-    df["perc_threshold_touched"] = (
-        bin_zonal_touched["sum"] / bin_zonal_touched["count"] * 100
-    )
-
-    return df
+    df_list = []
+
+    for a in gdf[bound_col].unique():
+        gdf_adm = gdf[gdf[bound_col] == a]
+
+        da_clip = raster_array.rio.set_spatial_dims(
+            x_dim=lon_coord, y_dim=lat_coord
+        ).rio.clip(gdf_adm[geom_col], all_touched=all_touched)
+        if stats_list is None:
```

Comment:
This should be initialized at the beginning of the method (and outside of the for loop) since it's setting the default